DOCSP-34286 -- Add Online Archive Create and Update JSON config options page (#249)
mongodb/docs-atlas-cli: source/reference/json/file-options-online-archive.txt

.. _atlas-cli-file-options-online-archive-file:

========================================
Online Archive Configuration File
========================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can use a configuration file to specify the required  
settings for creating and updating an online archive using the 
{+atlas-cli+}. The {+atlas-cli+} accepts ``.json`` online archive 
configuration files.

Use the following resources to:

- Learn the :ref:`required settings <required-oa-settings>` you 
  can specify in the online archive 

- View and copy :ref:`sample configuration files <example-oa-config-file>`.

.. _required-oa-settings:

Online Archive Settings
-----------------------

You can specify the following settings to create or 
update an online archive either in the configuration 
file or as flags in the command:

.. list-table:: 
   :header-rows: 1 
   :widths: 20 10 70 

   * - Field 
     - Type 
     - Description 

   * - ``groupId``
     - string
     - Unique identifier for your project. Your ``groupId`` is the same 

>        as your ``projectId``. For existing groups, your group/project id 

Reviewer: ```suggestion
       as your ``projectId``. For existing groups, your ``groupId`` and ``projectId``
```

       remains the same. The resource and corresponding 
       endpoints use the term groups.

   * - ``archiveID``
     - string
     - Unique 24-hexadecimal digit string that identifies 
       the online archive to update.
  
   * - ``clusterName``
     - string
     - Human-readable label that identifies the cluster that contains 
       the collection for which you want to create or update one online archive.

   * - ``collName``
     - string
     - The name of your collection.

   * - ``criteria``
     - object
     - Rules by which MongoDB MongoDB Cloud archives data.

   * - ``dbName``
     - string
     - Human-readable label of the database that 
       contains the collection that contains the online archive.

For detailed descriptions and a full list of available settings,

> see the request body schema in the API specification:

Reviewer: ```suggestion
see the request body schema in the API specification:

```
You need a blank line here for this to render as a bulleted list.

.. _atlas-cli-file-options-online-archive-file:

========================================
Online Archive Configuration File
========================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

You can use an Online Archive configuration file to specify the required 
settings for creating and updating an Online Archive. This archive stores 
data from one cluster within one project. To use this resource, the 
requesting API Key must have the Project Data Access Admin role.

Use the following resources to:

- Learn the required settings you can specify in the Online Archive 
  configuration file.

- View and copy sample configuration files.

Required Online Archive Create and Update Settings
--------------------------------------------------

> |service| requires the following settings to create an Online Archive with
> the {+atlas-cli+}. You must specify these Online Archive settings either
> in the configuration file or as flags in the command:

Reviewer: ```suggestion
You can specify the following settings to create or 
update an online archive either in the configuration 
file or as flags in the command:
```
I'm not sure if the first sentence is true for online atchive, but I think we should reword the 2nd sentence to indicate we can create or update.

.. list-table:: 
   :header-rows: 1 
   :widths: 20 10 70 

   * - Field 
     - Type 
     - Description 

   * - ``groupId``
     - string

>      - Unique identifier for your project. Your group id is the same 
>        as your project id. For existing groups, your group/project id 

Reviewer: ```suggestion
     - Unique identifier for your project. Your ``groupId`` is the same 
       as your ``projectId``. For existing groups, your ``groupId`` and ``projectId``
```
Suggestion: We should write out the parameter names so users know exactly what it is. I think their actual names are groupID and projectId, but feel free to update if not.

       remains the same. The resource and corresponding 
       endpoints use the term groups.

   * - ``archiveID``
     - string
     - Unique 24-hexadecimal digit string that identifies 

>        the online archive to update. This param is required for updating an
>        existing Online Archive.

Reviewer: ```suggestion
       the online archive to update.
```
Suggestion: I don't think the second sentence is needed since the first one states it's for updating an online archive.

  
   * - ``clusterName``
     - string

>      - Human-readable label that identifies the cluster that contains 
>        the collection for which you want to create one online archive.

Reviewer: ```suggestion
     - Human-readable label that identifies the cluster that contains 
       the collection for which you want to create or update one online archive.
```

   * - ``collName``
     - string
     - The name of your collection.

   * - ``criteria``
     - object
     - Rules by which MongoDB MongoDB Cloud archives data.

   * - ``dbName``
     - string
     - Human-readable label of the database that 
       contains the collection that contains the online archive.

> For a full list of available settings, see the API documentation for 
> :oas-atlas-op:`Create One Online Archive<>`.

Reviewer: ```suggestion
For detailed descriptions and a full list of available settings,
see the request body schema in the API specification:

- :oas-atlas-op:`Create One Online Archive </createOnlineArchive>`
- :oas-atlas-op:`Update One Online Archive </updateOnlineArchive>`
```

Example Online Archive Create Command
-------------------------------------

> After you create the file, run the command to create the Online Archive.

Reviewer: ```suggestion
To create an online archive configuration, specify the ``--file`` option 
and the path to the file. The following example creates an 
online archive by using a configuration file named 
``online-archive-config.json``:
```
Suggestion: Typically with commands, we write a statement about the params we're passing in, and what exactly we are doing.

Applies to the update section too.

.. code-block::

>     atlas clusters onlineArchive create --file online-archive-create.json --output json

Reviewer: ```suggestion
    atlas clusters onlineArchive create --file online-archive-config.json --output json
```
Suggestion: Lets rename the file to say `config` at the end instead of create -- I think we should omit the `create` in the file name since it's in the actual command, and may cause confusion to users.

Applies to the update section too.

Author: I think that it might be better to call it `online-archive-create-config.json` and `online-archive-update-config.json` because the configs are specific to those actions -- i.e. creating and updating an online archive respectively. WDYT?

Reviewer: That's fair -- I like your suggestion!

Example Online Archive Update Command
-------------------------------------
After you create the file, run the command to update the Online Archive.

.. code-block::

    atlas clusters onlineArchive update --file online-archive-update.json --output json

> Example Online Archive Create Configuration File

Reviewer: Suggestion: Structurally, I think we should show the example config file before the commands because you would need the file before you can run the command.
